### PR TITLE
AbstractSearchableSnapshotsRestTestCase.testClearCache() should wait for cache writes to complete

### PR DESCRIPTION
**Pull request by tlrx**
The test `AbstractSearchableSnapshotsRestTestCase.testClearCache()` sometimes fails because it assumes that all cache writes are completed when retrieving and comparing the searchable snapshots stat `cached_bytes_written`.

This commit changes the test so that it now waits for searchable snapshots thread pools to finish to process cache related tasks before retrieving the stats.

### CODE REVIEW

File: x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/AbstractSearchableSnapshotsRestTestCase.java [hunk 1]
```diff
@@ -458,6 +465,36 @@ protected static Number count(String index) throws IOException {
         return extractValue(responseAsMap(response), index + ".settings");
     }
 
+    @SuppressWarnings("unchecked")
+    protected static void waitForIdlingSearchableSnapshotsThreadPools() throws Exception {
```
Comment (original-brownbear) [inline x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/AbstractSearchableSnapshotsRestTestCase.java:469]: I wonder if we need to do it this complicated (although this is a little cleaner in terms of what we assert I guess), wouldn't we get the same effect from just extending the timeout on the busy assert for the cache size above?

**Review by original-brownbear — COMMENTED** [review]
One question on maybe doing a shorter fix.

File: x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/AbstractSearchableSnapshotsRestTestCase.java (hunk 1 above)
Comment (tlrx) [reply x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/AbstractSearchableSnapshotsRestTestCase.java:469]: I'd prefer to not extend timeouts as it does not make obvious why it is needed to wait here. I don't think this change is complicated and it mirrors what we already did at other places.

**Review by original-brownbear — APPROVED** [review]
LGTM :)

**Comment by tlrx** [comment]
Thanks Armin!

**Pull request merged**